app/models/track.rb

```diff
@@ -209,4 +223,20 @@ def valid_room
       errors.add(:room, "must be a room of #{program.conference.venue.name}")
     end
   end
+
+  ##
+  # Check that there is no other track in the same room with overlapping dates
+  def overlapping
+    return unless start_date && end_date && room && program.try(:tracks)
+    (program.tracks.accepted + program.tracks.confirmed - [self]).each do |other_track|
+      if other_track.room == room &&
+         other_track.start_date && other_track.end_date &&
+         (other_track.start_date <= start_date && other_track.end_date >= start_date ||
```

Comment:
I'd keep the same order of variables for all checks, and preferably I'd put the new track dates in the beginning, so that we can can read like this 'the start date (of the new track) is more or equal than the existing track'